toolkit/tools/pkg/imagecustomizerlib/versionsOfToolDependencies.go

```diff
@@ -0,0 +1,95 @@
+package imagecustomizerlib
+
+import (
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+)
+
+func getVersionsOfToolDeps() {
+	// Map of version flags with corresponding packages
+	versionFlags := map[string][]string{
+		"--version": {
+			"qemu-img", "rpm", "dd", "lsblk", "losetup", "sfdisk", "udevadm",
+			"flock", "blkid", "sed", "createrepo", "genisoimage", "parted", "mkfs",
+			"fsck", "fatlabel", "zstd", "veritysetup", "grub-install",
+		},
+		"-version": {
+			"mksquashfs",
+		},
+		"version": {
+			"openssl",
+		},
+		"-V": {
+			"mkfs.ext4", "mkfs.xfs", "e2fsck", "xfs_repair", "xfs_admin",
+		},
+		"none": {
+			"mkfs.vfat", "resize2fs", "tune2fs",
+		},
+	}
+
+	// Get distro and version
+	distro, version := getDistroAndVersion()
+	logger.Log.Debugf("Distro: %s, Version: %s\n", distro, version)
+
+	// Get versions of packages
+	logger.Log.Debugf("Tool Dependencies:")
+	for versionFlag, pkgList := range versionFlags {
+		for _, pkg := range pkgList {
+			version, err := getPackageVersion(pkg, versionFlag)
+			if err != nil {
+				logger.Log.Debugf("%s: not installed or error retrieving version\n", pkg)
+			} else {
+				logger.Log.Debugf("%s: %s", pkg, version)
+			}
+		}
+	}
+}
+
+// Function to get the distribution and version of host machine
+func getDistroAndVersion() (string, string) {
+	output, err := exec.Command("lsb_release", "-si").Output()
```

Comment:
`lsb_release` isn't default installed on Mariner (or Fedora). So, it probably isn't a good idea to rely on it.

It might be more reliable to pull the `NAME` and `VERSION` fields from `/etc/os-release`.

Reply:
Updated function to get the `NAME` and `VERSION` from `/etc/os-release`.